client/material/tooltip.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { useTransition, UseTransitionProps } from 'react-spring'
+import styled, { css, FlattenSimpleInterpolation } from 'styled-components'
+import { background900 } from '../styles/colors'
+import { caption } from '../styles/typography'
+import { OriginX, OriginY, PopoverContent, useAnchorPosition } from './popover'
+import { Portal } from './portal'
+import { shadow2dp } from './shadows'
+import { defaultSpring } from './springs'
+
+export type TooltipPosition = 'left' | 'right' | 'top' | 'bottom'
+
+const NoPointerPortal = styled(Portal)`
+  pointer-events: none;
+`
+
+const arrowStyle: Record<TooltipPosition, FlattenSimpleInterpolation> = {
+  left: css`
+    top: 50%;
+    right: 0px;
+
+    transform: translate(50%, -50%) rotate(45deg);
+    border-bottom: none;
+    border-left: none;
+  `,
+  right: css`
+    top: 50%;
+    left: 0px;
+
+    transform: translate(-50%, -50%) rotate(45deg);
+    border-top: none;
+    border-right: none;
+  `,
+  top: css`
+    bottom: 0px;
+    left: 50%;
+
+    transform: translate(-50%, 50%) rotate(45deg);
+    border-top: none;
+    border-left: none;
+  `,
+  bottom: css`
+    top: 0px;
+    left: 50%;
+
+    transform: translate(-50%, -50%) rotate(45deg);
+    border-bottom: none;
+    border-right: none;
+  `,
+}
+
+const TooltipContent = styled.div<{ $position?: TooltipPosition }>`
+  ${caption};
+  ${shadow2dp};
+
+  position: relative;
+  min-height: 24px;
+  padding: 4px 8px;
+
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  border: 1px solid rgba(255, 255, 255, 0.36);
+  border-radius: 4px;
+  background-color: ${background900};
+  pointer-events: none;
+
+  &::before {
+    content: '';
+    position: absolute;
+    width: 8px;
+    height: 8px;
+
+    background-color: inherit;
+    border: 1px solid rgba(255, 255, 255, 0.36);
+
+    ${props => arrowStyle[props.$position!]};
```

Comment:
If the intention is that this is always specified, you should just make it that way on the props. Otherwise, this should probably be `props.$position ? arrowStyle[props.$position] : ''` or something. (Since this isn't exported, I'd generally prefer just making the prop required)